src/views/exportJarFileUtil.ts

```diff
@@ -0,0 +1,336 @@
+        const outClassPaths: string[] = [];
+        const setUris: Set<string> = new Set<string>();
+        const pickDependencies: IJarQuickPickItem[] = [];
+        const pickedDependencies: IJarQuickPickItem[] = [];
+        for (const rootNode of rootNodes) {
+            const modulePaths: ClasspathResult = await extensionApi.getClasspaths(rootNode.uri, { scope: "runtime" });
+            generateDependencies(modulePaths.classpaths, setUris, pickDependencies, projectPath, true);
+            generateDependencies(modulePaths.modulepaths, setUris, pickDependencies, projectPath, true);
+            const modulePathsTest: ClasspathResult = await extensionApi.getClasspaths(rootNode.uri, { scope: "test" });
+            generateDependencies(modulePathsTest.classpaths, setUris, pickDependencies, projectPath, false);
+            generateDependencies(modulePathsTest.modulepaths, setUris, pickDependencies, projectPath, false);
+        }
+        if (pickDependencies.length === 0) {
+            return reject("No class path found.");
+        } else if (pickDependencies.length === 1) {
+            outClassPaths.push(pickDependencies[0].uri);
+            return resolve(outClassPaths);
+        }
+        const pickBox = window.createQuickPick<IJarQuickPickItem>();
+        pickDependencies.sort((node1, node2) => {
+            if (node1.description !== node2.description) {
+                return node1.description.localeCompare(node2.description);
+            }
+            if (node1.type !== node2.type) {
+                return node2.type.localeCompare(node1.type);
+            }
+            return node1.label.localeCompare(node2.label);
+        });
+        pickBox.items = pickDependencies;
+        pickDependencies.forEach((pickDependency) => {
+            if (pickDependency.picked) {
+                pickedDependencies.push(pickDependency);
+            }
+        });
+        pickBox.selectedItems = pickedDependencies;
+        pickBox.title = "Export Jar - Determine elements";
+        pickBox.placeholder = "Select the elements...";
+        pickBox.canSelectMany = true;
+        pickBox.ignoreFocusOut = true;
+        pickBox.buttons = pickSteps.length > 0 ? [(QuickInputButtons.Back)] : [];
+        pickBox.onDidTriggerButton((item) => {
+            if (item === QuickInputButtons.Back) {
+                reject(InputFlowAction.back);
+                pickBox.dispose();
+            }
+        });
+        pickBox.onDidAccept(() => {
+            pickBox.selectedItems.forEach((item) => {
+                outClassPaths.push(item.uri);
+            });
+            resolve(outClassPaths);
+            pickBox.dispose();
+        });
+        pickBox.onDidHide(() => {
+            reject();
+            pickBox.dispose();
+        });
+        pickBox.show();
+    });
+}
+
+function generateDependencies(paths: string[], setUris: Set<string>, pickDependencies: IJarQuickPickItem[],
```

Comment:
Instead of passing the `pickDependencies` into the method, I suggest you to return an array. And push the items outside the method:

```js
pickDependencies.push(...parseDependencyItems(...))
```